compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1095,17 +1095,7 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
     }
 
     fn pick_core(&self) -> Option<PickResult<'tcx>> {
-        let pick = self.pick_all_method(Some(&mut vec![]));
-
-        // In this case unstable picking is done by `pick_method`.
-        if !self.tcx.sess.opts.unstable_opts.pick_stable_methods_before_any_unstable {
-            return pick;
-        }
-
-        if pick.is_none() {
-            return self.pick_all_method(None);
-        }
-        pick
+        self.pick_all_method(Some(&mut vec![])).or_else(|| self.pick_all_method(None))
```

Comment:
With the context of `self.tcx.sess.opts.unstable_opts.pick_stable_methods_before_any_unstable` removed, it might worth adding a comment here:
```suggestion
		// Pick stable methods only first, and consider unstable candidates if not found.
        self.pick_all_method(Some(&mut vec![])).or_else(|| self.pick_all_method(None))
```